File: HammingDistance/Ruby/hamming_distance.rb
# take two binary strings and returns the Hamming Distance between them
def hamming_distance(string1, string2)
  if string1.length != string2.length
    return "Strings must be the same length."
  else
    total = 0
    for i in 0...string1.length
      if string1[i] != string2[i]
        total += 1
      end
    end
    return total
  end
end

File: algorithms/bit-manipulation/hamming-distance/c/HammingDistance.c
#include <stdio.h>

int HammingDistance(int a, int b) {
    unsigned int value = (unsigned int)(a ^ b);
    int distance = 0;
    while (value != 0U) {
        distance += (int)(value & 1U);
        value >>= 1U;
    }
    return distance;
}

int hamming_distance(int a, int b) {
    return HammingDistance(a, b);
}

int main(void) {
    printf("%d\n", HammingDistance(1, 4));
    return 0;
}
